### PR TITLE
chore(tests): migrate all remaining db integration tests

### PR DESCRIPTION
**Pull request by jpetto**
## Goal

migrate all remaining db integration tests (wow this was a CHORE).

- remove legacy db integration tests

there were only a 2 actual code changes here (commented in the PR) - the rest is just test migration/refactoring to API integration tests.

## Reference

Tickets:

- https://getpocket.atlassian.net/browse/BACK-1400

### CODE REVIEW

File: src/database/mutations/CollectionPartnerAssociation.ts [hunk 1]
```diff
@@ -103,9 +104,16 @@ export async function deleteCollectionPartnerAssociation(
...
   }
+
   // get the existing association for the internal id
   const association = await getCollectionPartnerAssociation(db, externalId);
 
+  if (!association) {
+    throw new UserInputError(
+      `Cannot delete a collection partner association with external ID "${externalId}"`
+    );
+  }
```
Comment (jpetto) [inline src/database/mutations/CollectionPartnerAssociation.ts:115]: this is one of 2 real code changes here.

File: src/database/mutations/CollectionStory.ts [hunk 2]
```diff
@@ -158,6 +159,12 @@ export async function deleteCollectionStory(
   // get the existing story for the internal id
   const existingStory = await getCollectionStory(db, externalId);
 
+  if (!existingStory) {
+    throw new UserInputError(
+      `Cannot delete a collection story with external ID "${externalId}"`
+    );
+  }
```
Comment (jpetto) [inline src/database/mutations/CollectionStory.ts:166]: this is 2 of 2 real code changes here.

Comment (kschelonka) [reply src/database/mutations/CollectionStory.ts:166]: does this need test coverage?

**Review by kschelonka — APPROVED** [review]

**Comment by kschelonka** [comment]
Checked out where you commented, the rest I'm going to trust :)

**Review by nina-py — APPROVED** [review]
Hooray! 🎉

**Pull request merged**